File: BEECROWD/2423_bolo_apisc2.py


class Bolo:
    def __init__(self):
        self.ingredientes = self.entrada_txt()
        self.ingredientes = self.cada_indice_inteiro(self.ingredientes)
        self.qtd_bolos = self.monitorar_ingredientes(self.ingredientes[0], self.ingredientes[1], self.ingredientes[2])
        print(self.qtd_bolos)
    
    @staticmethod
    def entrada_txt():
        entrada = str(input()).split(' ')
        return entrada
    
    @staticmethod
    def cada_indice_inteiro(iteravel):
        lista_de_inteiros = []
        for i in iteravel:
            lista_de_inteiros.append(int(i))
        return lista_de_inteiros
    
    @staticmethod
    def monitorar_ingredientes(a, b, c):
        farinha, ovos, leite = 2, 3, 5
        contador = 0
        
        # 4 6 9 -> 2 3 4
        while a >= farinha and b >= ovos and c >= leite:
            contador += 1
            a -= farinha
            b -= ovos
            c -= leite
        
        return contador

bolo = Bolo()

File: BEECROWD/2423_bolo_apisc2.js


/*
2 3 5  (quantidade p/ fazer um bolo)
4 6 10 (2) bolos
4 6 9  (1) bolos (por causa de 9/5 < 2)
*/

var input = require('fs').readFileSync('./dev/stdin', 'utf8');
var lines = input.split('\r').join('').split('\n');

function main() {
    const ingredientes = cada_indice_inteiro(tornar_lista(lines.shift(), ' '))
    const qtd_bolos = monitorar_ingredientes(ingredientes[0], ingredientes[1], ingredientes[2])
    
    exibir(qtd_bolos)
}

function exibir(conteudo) {
    console.log(conteudo)
}

function tornar_lista(alvo, separador) {
    return alvo.split(separador)
}

function tornar_inteiro(alvo) {
  return parseInt(alvo)
}

function cada_indice_inteiro(iteravel) {
  let lista_de_inteiros = []

  for (let i = 0; i < iteravel.length; i++) {
    lista_de_inteiros.push(tornar_inteiro(iteravel[i]))
  }
  return lista_de_inteiros
}

function eh_maior_igual(alvo, ref) {
  return alvo >= ref
}

function monitorar_ingredientes(item_a, item_b, item_c) {
  const [a, b, c] = [2, 3, 5]
  let contador = 0
  
  while (eh_maior_igual(item_a, a) && eh_maior_igual(item_b, b) && eh_maior_igual(item_c, c)) {
    item_a -= a
    item_b -= b
    item_c -= c
    contador++
  }
  return contador
}

main()
